src/easyreflectometry/sample/assemblies/gradient_layer.py

```diff
@@ -100,8 +101,8 @@ def roughness(self, roughness: float) -> None:
     def _dict_repr(self) -> dict[str, str]:
         """A simplified dict representation."""
         return {
-            'thickness': self.thickness,
-            'discretisation_elements': self._discretisation_elements,
+            'thickness': float(self.thickness),
+            'discretisation_elements': int(self._discretisation_elements),
```

Comment:
why are those explicit coversions needed?

Reply:
PR response
These are properties that else appears weird in the dictionary